lms/djangoapps/instructor/views/data_access.py

```diff
@@ -77,10 +77,24 @@ def get_group_query_students(course_id, group_id):
     Asynchronously makes the subqueries for a group and then aggregates them once the subqueries are finished.
     """
     _group, queries, _relation = get_saved_queries(course_id, group_id)
-    #wait for the queries to finish before proceeding
+    # wait for the queries to finish before proceeding
     make_subqueries.apply_async(args=(course_id, group_id, queries)).get()
-    queried_students = retrieve_grouped_query(course_id, group_id)
-    return queried_students
+    # forcing evaluation here because we will be deleting stuff afterward
+    subqueries_qs = GroupedTempQueryForSubquery.objects.filter(grouped_id=group_id).distinct().values_list('query_id',
+                                                                                                           flat=True)
+    if subqueries_qs:
+        subqueries = list(subqueries_qs)
+        queried_students_qs = retrieve_grouped_query(course_id, group_id)
+        if queried_students_qs:
+            queried_students = list(queried_students_qs)
+            #clear temporary table queries associated with these queries
+            old_queries = TemporaryQuery.objects.filter(id__in=subqueries)
+            saved_students = StudentsForQuery.objects.filter(query_id__in=old_queries.values_list(DatabaseFields.ID))
+            old_queries.delete()
+            saved_students.delete()
```

Comment:
Since you already take care of deleting orphaned students in the purge function, maybe we don't need to worry about deleting students here, and can just return the queryset?